openhcl/underhill_core/src/worker.rs

```diff
@@ -1736,7 +1744,7 @@ async fn new_underhill_vm(
         let manager = NvmeManager::new(
             &driver_source,
             processor_topology.vp_count(),
-            vfio_dma_buffer(&shared_vis_pages_pool),
+            vfio_dma_buffer(&shared_vis_pages_pool, "nvme_manager".into()),
```

Comment:
Maybe squeeze fixing this into this change? As you mention in the PR description, we need to push this down to be per device.

Reply:
Yep, let me see if i can do better here too.

Reply:
@chris-oo left a comment in https://github.com/microsoft/openvmm/pull/210 for you to take a look